packages/framework/ai-collab/src/explicit-strategy/agentEditReducer.ts

```diff
@@ -184,8 +195,29 @@ export function applyAgentEdit(
 			let insertedObject: TreeNode | undefined;
 			// if fieldSchema is a LeafnodeSchema, we can check that it's a valid type and set the field.
 			if (isPrimitive(modification)) {
-				// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-member-access
-				(node as any)[treeEdit.field] = modification;
+				try {
+					// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-member-access
+					(node as any)[treeEdit.field] = modification;
+				} catch (error) {
+					if (error instanceof UsageError === false) {
+						throw error;
+					}
+					// If the LLM attempts to use the wrong type for a field, we generate a useful error message that can be used as part of the feedback loop.
+					const isInvalidTypeError =
+						error.message.match(
```

Comment:
This is going to be pretty fragile - if the error message in tree is updated, this check will begin to silently fail. If we need to be able to match on this specific error, I recommend adding a new Error type to the tree package for this scenario specifically. Then we can against that kind of error in a way that should be detectable through API changes. We can make that error type `@internal` for now.

Reply:
I think the current unit tests would catch that scenario, since some of them check for the correct generation of `INVALID_TYPE` messages. Long-term I like the idea of more specific error types, but they'd probably need to come with type guard functions as well?

Reply:
I would still recommend against it. IMO we shouldn't rely on tests for anything we can reasonably cover using the type system.